src/evox/algorithms/mo/knea.py

```diff
@@ -0,0 +1,230 @@
+import jax
+import jax.numpy as jnp
+from functools import partial
+
+from evox import Algorithm, jit_class, State
+from evox.operators import (
+    selection,
+    mutation,
+    crossover,
+    non_dominated_sort,
+)
+from evox.utils import euclidean_dis
+from evox.operators.non_dominated_sort import _dominate_relation
+
+
+@partial(jax.jit, static_argnums=[4, 5])
+def mating_selection(key, pop, fit, knee, k, num_round):
```

Comment:
Move this function to `evox.operators.selection`
If it has a proper name, then use it, otherwise you use something like `knea_selection`.